File: leetcode/330/330.go
package main

func minPatches(nums []int, n int) int {
    var m, un, res, i, l = uint(1), uint(n), 0, 0, len(nums) 
    for m <= un {
        if i < l && uint(nums[i]) <= m {
            m += uint(nums[i])
            i++
        } else {
            m += m
            res++
        }
    }
    return res
}

File: leetcode/330/330.py
class Solution:
    def minPatches(self, nums: List[int], n: int) -> int:
        l = len(nums)
        m = 1
        i = 0
        res = 0
        while m <= n:
            if i < l and nums[i] <= m:
                m += nums[i]
                i += 1
            else:
                m += m 
                res += 1
        return res
